shader/fine.wgsl

```diff
@@ -63,13 +63,38 @@ var<storage> mask_lut: array<u32, 2048u>;
 let WG_SIZE = 64u;
 var<workgroup> sh_count: array<u32, WG_SIZE>;
 
-// This is 4 winding numbers packed to a u32, 8 bits per sample
-var<workgroup> sh_winding: array<atomic<u32>, 64u>;
-// Same packing, one group of 8 per pixel
-var<workgroup> sh_samples: array<atomic<u32>, SH_SAMPLES_SIZE>;
-// Same packing, accumulating winding numbers for vertical edge crossings
+// This array contains the winding number of the top left corner of each
+// 16 pixel wide row of pixels, relative to the top left corner of the row
+// immediately above.
+//
+// The values are stored packed, as 4 8-bit subwords in a 32 bit word.
+// The values are biased signed integers, with 0x80 representing a winding
+// number of 0, so that the range of -128 to 127 (inclusive) can be stored
+// without carry.
+//
+// For the even-odd case, the same storage is repurposed, so that a single
+// word contains 16 one-bit winding parity values packed to the word.
 var<workgroup> sh_winding_y: array<atomic<u32>, 4u>;
+// This array contains the winding number of the top left corner of each
+// 16 pixel wide row of pixels, relative to the top left corner of the tile.
+// It is expanded from sh_winding_y by inclusive prefix sum.
 var<workgroup> sh_winding_y_prefix: array<atomic<u32>, 4u>;
+// This array contains winding numbers of the top left corner of each
+// pixel, relative to the top left corner of the enclosing 16 pixel
+// wide row.
+//
+// During winding number accumulation, it stores a delta (winding number
+// relative to the pixel immediately to the left), then expanded using
+// prefix sum and reusing the same storage.
+//
+// The encoding and packing is the same as `sh_winding_y`. For the even-odd
+// case, only the first 16 values are used, and each word stores packed
+// parity values for one row of pixels.
+var<workgroup> sh_winding: array<atomic<u32>, 64u>;
+// This array contains winding numbers of multiple sample points within
+// a pixel, relative to the winding number of the top left corner of the
+// pixels. The encoding and packing is the same as `sh_winding_y`.
```

Comment:
> pixels. The encoding...

of the pixel (singular) or is this relative to a group of pixels?

Reply:
I went back and forth between singular and plural. Fixed.